Update provisioner

tborealis opened the pull request:
- Switch from Debian Jessie to Stretch
- Update PHP to 7.1
- Improve provisioning performance
- Clean up Ansible warnings and update roles
- Improve CI performance and logging
- Increase CI coverage by running yarn, webpack and website builds
- Add PHP linting
- Fix shorewall role and reinstate

tborealis committed:
Upgrade Vagrant box to Debian Stretch

tborealis committed:
Fix a Linux NFS issue by forcing TCP for NFS

tborealis committed:
Update base role from upstream

tborealis committed:
Move ansible config to ansible folder

tborealis committed:
Update nginx role from upstream

tborealis committed:
Update to PHP 7.2

- Remove dotdeb as it has been deprecated and replace with Sury repo
- Update PHP role from upstream
- Improve xdebug config

tborealis committed:
Update composer role from upstream

tborealis committed:
Update friendsofphp/php-cs-fixer to version that supports PHP 7.2

tborealis committed:
PHP CS fixes

tborealis committed:
Set ansible compatibility_mode to remove provision warning

tborealis committed:
Remove ansible app task as empty and causing warning

tborealis committed:
Fix deprecated include warning

tborealis committed:
Update node role from upstream

- Fixes "creates" warning and prevents setup script run on all provisions
- Uses repo key for better security

tborealis committed:
JUnit output from PHPUnit

tborealis committed:
Specify node version used in CircleCI

tborealis committed:
Update PHP version in CircleCI

Not the 7.2.* it should be to match the current local env but closer than 7.0.*

tborealis committed:
Update CircleCI to be closer to full build

- Override dependencies to avoid npm install (we use yarn)
- Generate website and run encore as part of tests

tborealis committed:
Incorrect version of PHP used on CircleCI due to absolute path usage

tborealis committed:
Fix composer caching on CircleCI

tborealis committed:
Merge branch 'develop' into feature/update-provisioner

tborealis committed:
CS fixes

tborealis committed:
Add PHP linting

tborealis committed:
Update shorewall role from upstream and reinstate

DaveLiddament commented on ansible/roles/base/templates/etc/ssh/sshd_config.j2:
```diff
@@ -60,7 +60,7 @@ StrictModes yes
 SyslogFacility AUTH
 TCPKeepAlive no
-UseDNS yes
+UseDNS no
```
QUESTION: What does this mean?

tborealis replied:
This controls whether sshd does a reverse DNS lookup when you log in. It slows down login and is only really needed if you control access using hostname. See https://unix.stackexchange.com/a/56947 and the answer after that one.

DaveLiddament commented on ansible/group_vars/all.yml:
```diff
@@ -19,12 +19,17 @@ default_users:
+- php7.2-dom
+- php7.2-mbstring
+
+composer_php_version: 7.2
```
QUESTION: Does this relate directly to PHP version? Should it be separate from php_cli_version?

tborealis replied:
Composer has a dependency on the php zip module. This ensures that it installs the right module as version number is a part of the package name. I think it would be better to default this to the CLI version number as that will always be what you are installing it against. I'll update this.

DaveLiddament commented on ansible/roles/composer/tasks/main.yml:
```diff
@@ -12,8 +18,11 @@
-- include: vagrant.yml
-  when: vagrant
+- include_tasks: vagrant.yml
+  when: vagrant and composer_create_swap
```
Is this still required? I thought composer had fixed this?

tborealis replied:
It has as far as I know. I have left it in and defaulted composer_create_swap to false in case it becomes an issue again.

DaveLiddament commented on ansible/group_vars/vagrant.yml:
```diff
@@ -6,13 +6,15 @@ app_user: vagrant
   dest: phpsw_dev
+  enabled: true
 
 ssl_keys:
```
I know not specific to this current review, but what are these SSL keys doing?

tborealis replied:
No idea. I'll remove them when considering the SSL part.

DaveLiddament reviewed (commented):
We either need circle CI to be running 7.2 or downgrade PHP version to 7.1. Both code and CircleCI must be running same version of PHP.

tborealis committed:
Update base role from upstream

- Fix hostname changed on every run

tborealis committed:
Update ssl role from upstream

- Use modern layout
- Use arrays rather than structs for params
- Remove unused self signed key and cert

tborealis committed:
Update shorewall role from upstream

- Trigger restart on install of shorewall

tborealis committed:
Update git role from upstream

- Use modern layout

tborealis committed:
Update php and composer roles from upstream

- Use common php versions across roles
- Update group_vars to support

tborealis committed:
Revert php version to 7.1 (CircleCI doesn't support 7.2)

tborealis committed:
Merge remote-tracking branch 'origin/develop' into feature/update-provisioner

tborealis commented:
I went for PHP 7.1 as CircleCI doesn't support 7.2 yet

tborealis committed:
Fix CS issues

DaveLiddament committed:
Merge branch 'develop' into feature/update-provisioner
